packages/schemas/alterations/next-1669091623-roles-and-scopes.ts

```diff
@@ -0,0 +1,47 @@
+import { sql } from 'slonik';
+
+import type { AlterationScript } from '../lib/types/alteration.js';
+
+const alteration: AlterationScript = {
+  up: async (pool) => {
+    await pool.query(sql`
+      -- scopes
+      create table scopes (
+        id varchar(21) not null,
+        resource_id varchar(21) references resources (id) on update cascade on delete cascade,
+        name varchar(256) not null,
+        description text,
+        created_at timestamptz not null default(now()),
+        primary key (id)
+      );
+      -- update table roles, add id and replace pkey
+      alter table roles add column id varchar(21);
+      update roles set id = name;
+      alter table roles alter column id set not null;
+      alter index roles_pkey rename to roles_pkey_1;
+      create unique index roles_pkey on roles using btree(id)
+      drop index roles_pkey_1;
```

Comment:
no need for this?

Reply:
Need to change prev "primary key index" to a normal unique index, updated with more straitforword queries. PTAL